src/game/client/components/menus_settings.cpp

```diff
@@ -566,6 +566,90 @@ class CLanguage
 	bool operator<(const CLanguage &Other) { return m_Name < Other.m_Name; }
 };
 
+
+int CMenus::ThemeScan(const char *pName, int IsDir, int DirType, void *pUser)
+{
+	CMenus *pSelf = (CMenus *)pUser;
+	int l = str_length(pName);
+
+	if(l < 4 || IsDir || str_comp(pName+l-4, ".map") != 0)
+		return 0;
+	char aFullName[128];
+	char aThemeName[128];
+	str_copy(aFullName, pName, min((int)sizeof(aFullName),l-3));
+
+	l = str_length(aFullName);
+	bool isDay = false;
+	bool isNight = false;
+	if(l > 4 && str_comp(aFullName+l-4, "_day") == 0)
+	{
+		str_copy(aThemeName, pName, min((int)sizeof(aThemeName),l-3));		
+		isDay = true;
+	}
+	else if(l > 6 && str_comp(aFullName+l-6, "_night") == 0)
+	{
+		str_copy(aThemeName, pName, min((int)sizeof(aThemeName),l-5));		
+		isNight = true;
+	}
+	else
+		return 0;
+	
+	// try to edit an existing theme
+	for(sorted_array<CTheme>::range r = pSelf->m_lThemes.all(); !r.empty(); r.pop_front()) // bit slow but whatever
```

Comment:
for(int i = 0; i < pSelf->m_lThemes.size(), ++i)
{

	if(str_comp(pSelf->m_lThemes[i].m_aName, aThemeName) == 0)

	{...